### ISSUE TITLE
Custom pluralization rule is not working

### ISSUE THREAD

**Issue by RollingTL** [intlify/vue-i18n-next#882]
### Reporting a bug?

Hi,
I am trying to use custom pluralization function but it seems it is not working as expected:

The result is:
```
<p>1 машина</p>
<p>2 машины</p>
<p>4 машины</p>
<p>12 машины</p>
<p>21 машины</p>
```

### Expected behavior

```
<p>1 машина</p>
<p>2 машины</p>
<p>4 машины</p>
<p>12 машин</p>
<p>21 машина</p>
```

12 and 21 result is wrong.

### Reproduction

[https://github.com/RollingTL/Testing-intlify-vue-i18n-next-custom-pluralization](url)

### System Info

```shell
Mac OS
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/intlify/vue-i18n-next/blob/master/.github/CONTRIBUTING.md)
- [X] Read the [Documentation](https://vue-i18n.intlify.dev/)
- [X] Check that there isn't [already an issue](https://github.com/intlify/vue-i18n-next/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/intlify/vue-i18n-next/discussions)
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by RollingTL** [comment]
I played with pluralization function and I think that it never called.
And one question what is the purpose of second and third argument? ()choicesLength, orgRule)

**Comment by RollingTL** [comment]
After reading sources I have found the problem 
Instead of **pluralizationRules**  as in documentation the option should be **pluralRules**.


```
const i18n = createI18n({
  locale: 'ru',
  // the custom rules here ...
  **pluralRules**: {
    ru: customRule
  },
  messages: {
    ru: {
      car: '0 машин | {n} машина | {n} машины | {n} машин',
      banana: 'нет бананов | {n} банан | {n} банана | {n} бананов'
    }
  }
})
```
It is necessary to change documentation.

**Comment by kazupon** [comment]
Thank you for your feedback.

About custom pluralization, vue-i18n@v9.x or later have a breaking change 
You need to check docs here:
https://vue-i18n.intlify.dev/guide/migration/breaking.html#pluralization-api-return-value

In composition API, we don't mention custom pluralization.
We should improve the docs about custom pluralization.
https://vue-i18n.intlify.dev/guide/advanced/composition.html#pluralization

**Comment by FADL285** [comment]
The problem is still found instead of '**pluralizationRules**' as in documentation it should be '**pluralRules**'

or change the code to be **pluralizationRules**

**Comment by mewforest** [comment]
OMG There is still an issue in documentation! 😮

> The problem is still found instead of '**pluralizationRules**' as in documentation it should be '**pluralRules**'

**Comment by nulen** [comment]
Yup, please fix it, chat gpt also suggests "pluralizationRules"

**Comment by BobbieGoede** [comment]
As [described in the documentation](https://vue-i18n.intlify.dev/guide/essentials/pluralization#custom-pluralization:~:text=To%20use%20the,for%20Composition%20API) both `pluralizationRules` and `pluralRules` are the correct keys, it just depends on whether you're using the Composition API or the Options API.

* `pluralizationRules` (Options/legacy API)
* `pluralRules` (Composition API)

See https://vue-i18n.intlify.dev/guide/essentials/pluralization.html#custom-pluralization and compare [highlighted lines in breaking changes](https://vue-i18n.intlify.dev/guide/migration/breaking#pluralization-api-return-value:~:text=pluralizationRules%3A%20%7B,%7D%2C) with [highlighted lines in breaking changes](https://vue-i18n.intlify.dev/guide/migration/breaking#pluralization-api-return-value:~:text=pluralRules%3A%20%7B,%7D%2C).

If you feel like this could be documented in a better way feel free to open a PR!

>Yup, please fix it, chat gpt also suggests "pluralizationRules"

As for this, we have no control over what information chat GPT uses and it's your own responsibility to verify its output.

**Comment by Divensky** [comment]
@RollingTL earlier asked about the purpose of second and third argument (choicesLength, orgRule). 

I found an [explanation of the choicesLength](https://kazupon.github.io/vue-i18n/guide/pluralization.html#custom-pluralization) as an overall amount of available choices. But I did not find an explanation of `orgRule` anywhere. 

Could anyone help with this explanation? I wonder if it might help to customize rule sets for different words. Within a Slavic language, different words have different plural forms... choicesLength might help as workaround but I wonder if there is a direct way to set a rule for a specific group of words in that locale.
